jax/_src/lax/parallel.py

```diff
@@ -1124,7 +1124,9 @@ def source_to_front(group):
   replica_groups = [source_to_front(group) for group in replica_groups]
   channel = ctx.module_context.new_channel()
   return hlo.CollectiveBroadcastOp(
-      x, replica_groups=_replica_groups_hlo(replica_groups)).results
+      x, replica_groups=_replica_groups_hlo(replica_groups),
```

Comment:
I think you need to follow this pattern: 
![image](https://github.com/user-attachments/assets/f5c46ebb-1716-4cfa-92e2-5d4a58b7a6e0)

Reply:
where `is_spmd = isinstance(axis_context, (SPMDAxisContext, ShardingContext))`

Reply:
OK. Yeah, this is some old problem of xla. I will follow this.

Reply:
Done. Please take a look! 

I think this thing has spread in many collective lowering function and probably deserve a clean up refactoring. But it can be done separately.